lib/events/api.go

```diff
@@ -351,10 +351,15 @@ const (
 	TrustedClusterCreateEvent = "trusted_cluster.create"
 	// TrustedClusterDeleteEvent is the event for removing a trusted cluster.
 	TrustedClusterDeleteEvent = "trusted_cluster.delete"
-	// TrustedClusterTokenCreateEvent is the event for
-	// creating new join token for a trusted cluster.
+	// TrustedClusterTokenCreateEvent is the event for creating new provisioning
+	// token for a trusted cluster. Deprecated in favor of
+	// [ProvisionTokenCreateEvent].
 	TrustedClusterTokenCreateEvent = "trusted_cluster_token.create"
 
+	// ProvisionTokenCreateEvent is the event for creating a provisioning token,
+	// also known as Join Token. See [types.ProvisionToken].
+	ProvisionTokenCreateEvent = "join_token.create"
```

Comment:
Why are we using two different nomenclatures for this concept? Is that already the case elsewhere in the codebase/docs? Could we simplify?

Reply:
> Why are we using two different nomenclatures for this concept? Is that already the case elsewhere in the codebase/docs? Could we simplify?

Internally, we seem to refer to the Resource as "ProvisionToken" (e.g the type is called ProvisionTokenV2) - however - almost all external references to this call it a "join token" or just "token". I advised Bartosz to name this event "join_token.create" externally for consistency with what customers are used to - and at some point in the future - we'll rename any references to `ProvisionToken` to `JoinToken` internally (it's shorter, and much more understandable).

Reply:
(Also: I also considered going with `JoinToken` in code right now, but then, the naming issue is deeper than that; the entire "provisioning" term stands in the way, and I'm just not ready for this ATM.)